pkg/api-server/resource_endpoints.go

```diff
@@ -418,7 +419,7 @@ func (r *resourceEndpoints) validateResourceRequest(request *restful.Request, re
 	if name != resourceMeta.Name {
 		err.AddViolation("name", "name from the URL has to be the same as in body")
 	}
-	if r.mode == config_core.Zone && !r.doesNameLengthFitsGlobal(name) {
+	if r.mode == config_core.Zone && r.federatedZone && !r.doesNameLengthFitsGlobal(name) {
```

Comment:
Can we just remove the r.mode?

Reply:
Actually should this always be the case (even in non federated cases)?

Reply:
Do we also have a future workaround this limitation with the name stuff from Ilya?

Reply:
we can, yes

Reply:
> Do we also have a future workaround this limitation with the name stuff from Ilya?

Yes, we won't append `.zone` suffix in the future (it will be replaced with hash-suffix + labels). This means won't don't need `doesNameLengthFitsGlobal` func